.github/workflows/build_packages.yml

```diff
@@ -34,18 +43,35 @@ jobs:
       - name: Install Python packages
         run: pip install packaging
 
+      # Compute version suffix based on inputs (default to 'rc')
+      - name: Compute stable version suffix
+        if: ${{ inputs.build_type == 'stable' }}
+        run: |
+          version_suffix=""
+          echo "version_suffix=${version_suffix}" >> $GITHUB_ENV
+      - name: Compute rc version suffix
+        if: ${{ inputs.build_type == 'rc' || inputs.build_type == '' }}
+        run: |
+          version_suffix="$(printf 'rc%(%Y%m%d)T')"
+          echo "version_suffix=${version_suffix}" >> $GITHUB_ENV
+      - name: Compute dev version suffix
+        if: ${{ inputs.build_type == 'dev' }}
+        run: |
+          version_suffix=".dev0+${{ github.sha }}"
+          echo "version_suffix=${version_suffix}" >> $GITHUB_ENV
```

Comment:
We could also move this logic into `compute_local_version.py`, as it is in IREE: https://github.com/iree-org/iree/blob/ab3c9bb031f82f8481da7d867fab03db9180646a/build_tools/python_deploy/compute_local_version.py#L26-L30

We'd then need to pass the selection from the workflow inputs through to the script somehow:

```yml
      - name: Generate release candidate versions
        id: version_rc
        run: |
          python3 build_tools/python_deploy/compute_local_version.py --release-type ${{ inputs.build_type }} sharktank
          # extract version_suffix from the local version to forward to the job output?
          # that is currently passed to `write_requirements.py` down below as part of building the shark-ai package
```

```python
release_type = parser.add_mutually_exclusive_group()
release_type.add_argument("-stable", "--stable-release", action="store_true")  # default
release_type.add_argument("-rc", "--nightly-release", action="store_true")
release_type.add_argument("-dev", "--development-release", action="store_true")
release_type.add_argument("--custom-string", action="store", type=str)
release_type.add_argument("--build-type", action="store", type=str)

args = parser.parse_args()
if args.build_type == "dev":
  args.development_release = True
elif args.build_type == "stable
  ...
```

I generally prefer to keep workflow files simpler and put logic in scripts that can be run outside of the CI. Feeling lazy about this though, since it is already working as-is :P. Opinions?

Reply:
Well I would like to harmonize what we have here and what I've done for IREE in general. However, I am fine with going what you have right now (assuming that one of us fills an issue, see my review comment about refactoring to reduce confusion).